vision/api/label/snippets.py

```diff
@@ -0,0 +1,98 @@
+
+# Copyright 2016 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import base64
+import httplib2
+import json
+
+from googleapiclient import discovery
+from oauth2client.client import GoogleCredentials
+
+DISCOVERY_URL = ('https://vision.googleapis.com/$discovery/rest?'
+                 'labels=TRUSTED_TESTER&version=v1')
+
+
+def get_service():
+    """Get vision service using discovery."""
+    credentials = GoogleCredentials.get_application_default()
+    scoped_credentials = credentials.create_scoped(
+        ['https://www.googleapis.com/auth/cloud-platform'])
+    http = httplib2.Http()
+    scoped_credentials.authorize(http)
+    return discovery.build('vision', 'v1',
+                           http=http,
+                           discoveryServiceUrl=DISCOVERY_URL)
+
+
+def crop_hint(photo_path):
+    """Run a crop hint request on the image."""
+
+    service = get_service()
+
+    with open(photo_path, 'rb') as image:
+        image_content = base64.b64encode(image.read())
+        service_request = service.images().annotate(body={
+            'requests': [{
+                'image': {
+                    'content': image_content.decode('UTF-8')
+                },
+                'features': [{
+                    'type': 'CROP_HINTS'
+                }]
+            }]
+        })
+
+        response = service_request.execute()
+        return response
+
+
+def web_annotation(photo_path):
+    """Run a web annotation request on the image."""
+
+    service = get_service()
+
+    with open(photo_path, 'rb') as image:
+        image_content = base64.b64encode(image.read())
+        service_request = service.images().annotate(body={
+            'requests': [{
+                'image': {
+                    'content': image_content.decode('UTF-8')
+                },
+                'features': [{
+                    'type': 'WEB_ANNOTATION',
+                    'maxResults': 10
+                }]
+            }]
+        })
+
+        response = service_request.execute()
+        return response
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('command', choices=['crop_hint', 'web_annotation'])
+    parser.add_argument('image_path', help='The image you\'d like to process.')
+    args = parser.parse_args()
+
+    if args.command == 'crop_hint':
+        response = crop_hint(args.image_path)
+        print(json.dumps(response, indent=2))
```

Comment:
Our samples should not be structured around our tests, it should be the other way around. Please be consistent with the other samples.